project-ivi/ivi
# Linting

\#108

# Description

Added linting and linting-related workflows. Follows recommended eslint config with several custom rules to enforce **no semi, single quotes, 2 space indent** among others. Also a few react-specific config so jsx is properly linted.

# Change Log
Describe all changes made 

- `eslint.json` : New file - added basic config with several custom rules
- `package.json` : added `lint` and `lint-w` scripts
- `.travis.yml` : added linting to travis process
- `README.md` : added how to lint section, fixed one quote
- `src/...` : all src files are free of linting errors. 

# Manual Test Steps
1. `npm install`
2. `npm run lint` - Notice: no errors
3. Make a style error (change `'` to `"` somewhere, indent too much...)
4. run `./node_modules/.bin/eslint src/` - Notice: your error is reported
5. `npm run lint` - Notice: no errors, Notice: your error has been corrected in the file
6. `npm run lint-w` starts auto-lint and updates when file changed
